### ISSUE TITLE
await (async () => {})() :: Cannot find name 'await'.ts(2304)

### ISSUE THREAD

**Issue by teknsl** [denoland/deno#5918]
```ts
// working
const asyncFunc = async () => {
    return someValue;
}
let someValue = await asyncFunc();

// prefered but not working
let someValue = await (async () => {
   return someValue
})();
```
syntactically it is correct, but still it returns error ``Cannot find name 'await'.ts(2304)``

**Comment by marcosc90** [comment]
Note: It works perfectly fine with `.js` files, it's the TS compiler throwing that error.

**Comment by kitsonk** [comment]
I thought we had an issue for it open here, but it is an upstream TypeScript bug.  Ref: https://github.com/microsoft/TypeScript/issues/38483

**Comment by MarkTiedemann** [comment]
@kitsonk You mentioned this issue before in https://github.com/denoland/deno/issues/471, which is a general issue about top-level await. Might be good to keep this one open just for this specific issue.

**Comment by jingkaimori** [comment]
this error disappear on deno 1.19.1, can this issue be closed?

**Comment by soshimee** [comment]
Not fixed on Deno `1.20.3`. At least on Windows.

**Comment by kitsonk** [comment]
@soshimee can you please provide a full reproduction and the full output of `deno --version`.

**Comment by MarkTiedemann** [comment]
Can confirm. This never worked.

With `v1.19.2`:

```
C:\dev\test>.\deno.exe run repro.ts
error: TS2304 [ERROR]: Cannot find name 'await'.
await (async () => 1)();
~~~~~
    at file:///C:/dev/test/repro.ts:1:1
```

With `v1.21.2`:

```
C:\dev\test>.\deno.exe run repro.ts
Check file:///C:/dev/test/repro.ts
error: TS2311 [ERROR]: Cannot find name 'await'. Did you mean to write this in an async function?
await (async () => 1)();
~~~~~
    at file:///C:/dev/test/repro.ts:1:1
```

`repro.ts` containing `await (async () => 1)();`.

**Comment by nhrones** [comment]
works for me
```
deno 1.21.2 (release, x86_64-pc-windows-msvc)
v8 10.0.139.17
typescript 4.6.2
\C:\Users\nhron> deno
Deno 1.21.2
exit using ctrl+d or close()
> await (async () => 1)();
1
>
```

**Comment by MierenManz** [comment]
the problem is not the code execution but the typescript error it raises when trying to typecheck the file

**Comment by nayeemrmn** [comment]
Fixed by #14875.
